staging/src/k8s.io/apimachinery/pkg/runtime/extension.go

```diff
@@ -25,7 +26,10 @@ func (re *RawExtension) UnmarshalJSON(in []byte) error {
 	if re == nil {
 		return errors.New("runtime.RawExtension: UnmarshalJSON on nil pointer")
 	}
-	re.Raw = append(re.Raw[0:0], in...)
+
+	if !bytes.Equal(in, []byte("null")) {
+		re.Raw = append(re.Raw[0:0], in...)
```

Comment:
I added a test to verify the marshal and unmarshal will round trip correctly.  If that test is correct then this does properly round trip